src/miner_lora.erl

```diff
@@ -278,27 +322,36 @@ handle_cast(_Msg, State) ->
     lager:warning("rcvd unknown cast msg: ~p", [_Msg]),
     {noreply, State}.
 
-handle_info(reg_domain_timeout, #state{reg_domain_confirmed=false, pubkey_bin=Addr} = State) ->
-    lager:debug("checking regulatory domain for address ~p", [Addr]),
+handle_info(chain_check, State) ->
+    case blockchain_worker:blockchain() of
+        undefined ->
+            erlang:send_after(500, self(), chain_check),
+            {noreply, State};
+        Chain ->
+            ok = blockchain_event:add_handler(self()),
+            {noreply, update_state_using_chain(Chain, State)}
+    end;
+handle_info({blockchain_event, {new_chain, NC}}, State) ->
+    {noreply, update_state_using_chain(NC, State)};
+handle_info({blockchain_event, _}, State) ->
+    {noreply, State};
+handle_info(reg_domain_timeout, #state{chain=undefined} = State) ->
+    %% There is no chain, we cannot lookup regulatory domain data yet
+    %% Keep waiting for chain
+    erlang:send_after(500, self(), chain_check),
+    {noreply, State};
+handle_info(reg_domain_timeout, #state{reg_domain_confirmed=false, pubkey_bin=Addr, chain=Chain} = State) ->
+    lager:info("checking regulatory domain for address ~p", [Addr]),
     %% dont crash if any of this goes wrong, just try again in a bit
     try
-        case ?MODULE:reg_domain_data_for_addr(Addr) of
-            {error, Reason}->
-                lager:debug("cannot confirm regulatory domain for miner ~p, reason: ~p", [Reason]),
-                %% the hotspot has not yet asserted its location, transmits will remain disabled
-                %% we will check again after a period
-                erlang:send_after(30000, self(), reg_domain_timeout),
-                {noreply, State};
-            {ok, {Region, FrequencyList}} ->
-                lager:info("confirmed regulatory domain for miner ~p.  region: ~p, freqlist: ~p",
-                    [Addr, Region, FrequencyList]),
-                {noreply, State#state{ reg_domain_confirmed = true, reg_region = Region,
-                        reg_freq_list = FrequencyList, reg_throttle = miner_lora_throttle:new(Region)}}
-        end
+        TempState = maybe_update_reg_data(State),
+        Throttle = miner_lora_throttle:new(reg_region(TempState)),
+        NewState = TempState#state{reg_throttle=Throttle},
+        {noreply, NewState}
     catch
         _Type:Exception ->
-            lager:warning("error whilst checking regulatory domain: ~p.  Will try again...",[Exception]),
-            erlang:send_after(30000, self(), reg_domain_timeout),
+            lager:warning("error whilst checking regulatory domain: ~p.  chain: ~p. Will try again...", [Exception, Chain]),
+            erlang:send_after(1000, self(), reg_domain_timeout),
```

Comment:
Why is this so aggressive now?

Reply:
Made it so its only aggressive when testing.